src/Equilibrium/EquilibriumTypes.jl

```diff
@@ -0,0 +1,339 @@
+#=
+This file is the one stop shop for all the fundemental structures used in 
+    creating equilibrium descriptions for the DCON ODE to use.
+=#
+
+using Base: @kwdef
+
+# --- Helper function --- 
+
+
+function symbolize_keys(dict::Dict{String, Any})
+    return Dict(Symbol(k) => v for (k, v) in dict)
+end
+
+
+# --- Main Structures for the Equilibrium Code ---
+
+
+@kwdef mutable struct EquilControl
+    eq_type::String = "efit"
+    eq_filename::String = "mypath"
+
+    jac_type::String = "hamada"
+    power_bp::Int = 0
+    power_b::Int = 0
+    power_r::Int = 0
+
+    grid_type::String = "ldp"
+    psilow::Float64 = 1e-2
+    psihigh::Float64 = 0.994
+    mpsi::Int = 128
+    mtheta::Int = 256
+
+    newq0::Int = 0
+    etol::Float64 = 1e-7
+    use_classic_splines::Bool = false
+
+    input_only::Bool = false
+    use_galgrid::Bool = true
+
+    """
+    Modified internal constructor that enforces self consistency within the inputs
+    """
+    function EquilControl(eq_type, eq_filename, jac_type, power_bp, power_b, power_r,
+        grid_type, psilow, psihigh, mpsi, mtheta, newq0, etol, use_classic_splines,
+        input_only,use_galgrid)
+        if jac_type == "hamada"
+            @info "Forcing hamada coordinate jacobian exponents: power_*"
+            power_b=0
+            power_bp=0
+            power_r=0
+        elseif jac_type == "pest"
+            @info "Forcing pest coordinate jacobian exponents: power_*"
+            power_b=0
+            power_bp=0
+            power_r=2
+        elseif jac_type == "equal_arc"
+            @info "Forcing equal_arc coordinate jacobian exponents: power_*"
+            power_b=0
+            power_bp=1
+            power_r=0
+        elseif jac_type == "boozer"
+            @info "Forcing boozer coordinate jacobian exponents: power_*"
+            power_b=2
+            power_bp=0
+            power_r=0
+        elseif jac_type == "park"
+            @info "Forcing park coordinate jacobian exponents: power_*"
+            power_b=1
+            power_bp=0
+            power_r=0
+        elseif jac_type == "other"
+            @info "Using manual jacobian exponents: power b, bp, r = $(power_b), $(power_bp), $(power_r)"
+        elseif jac_type != "other"
+            error("Cannot recognize jac_type = $(jac_type)")
+        end
```

Comment:
I didn't realize that most of the `jac_type` options forced this. You might make outer constructors that don't take `power_b`, `power_bp`, or `power_r` as inputs and automatically fill these out from `jac_type`. 

You could also make a `const` dictionary that does this mapping for you